core/src/main/java/org/polypheny/db/docker/DockerInstance.java

```diff
@@ -269,24 +271,39 @@ boolean hasContainers() throws IOException {
     }
 
 
-    void updateConfig( String host, String alias, String registry ) {
-        throw new NotImplementedException( "Updating configurations has been temporarily disabled" );
-        /*
+    Optional<HandshakeInfo> updateConfig( String hostname, String alias, String registry ) {
         synchronized ( this ) {
-            if ( !this.host.hostname().equals( host ) ) {
+            DockerHost newHost = new DockerHost( hostname, alias, registry, this.getHost().communicationPort(), this.getHost().handshakePort(), this.getHost().proxyPort() );
+            if ( !this.host.hostname().equals( hostname ) ) {
                 client.close();
-                this.host = host;
                 status = Status.NEW;
+//                try {
+//                    PolyphenyKeypair kp = PolyphenyCertificateManager.loadClientKeypair( "docker", host.hostname() );
+//                    byte[] serverCertificate = PolyphenyCertificateManager.loadServerCertificate( "docker", host.hostname() );
+//                    PolyphenyDockerClient client = new PolyphenyDockerClient( newHost.hostname(), newHost.communicationPort(), kp, serverCertificate );
+//                    client.ping();
+//                    if ( !client.getDockerId().equals( dockerInstanceUuid ) ) {
+//
+//                    }
+//                    // Getting here means everything is OK.
+//                } catch ( IOException e ) {
+//                    log.error( "Failed to use new Docker instance", e );
+//                    throw new DockerUserException( "Cannot update to Docker instance" );
+//                }
+//
+                // TODO: Copy/Move keys...
+                // TODO: Restart all proxy connections
```

Comment:
Open todo

Reply:
This is an opportunity for future improvements